File: src/main/sources/com/hebangdata/daos/Probability.scala
package com.hebangdata.daos

/**
  * 词语的出现次数实体类
  * @param initial
  * @param letter
  * @param count_0
  * @param count_1
  * @param count_2
  * @param count_3
  */
final case class Probability
(
  initial: String,
  letter: String,
  count_0: java.math.BigDecimal,
  count_1: java.math.BigDecimal,
  count_2: java.math.BigDecimal,
  count_3: java.math.BigDecimal
) {

}

trait ProbabilityMapper {
  def find(initial: String, letter: String): Probability
  def insert(entity: Probability): Int
  def delete(): Unit
}

File: src/main/sources/com/hebangdata/daos/Probability.java
package com.hebangdata.daos;

/**
 * 词语的出现次数实体类
 */
public class Probability {
	public String initial;
	public String letter;
	public java.math.BigDecimal count_0;
	public java.math.BigDecimal count_1;
	public java.math.BigDecimal count_2;
	public java.math.BigDecimal count_3;

	public Probability(
			final String initial, final String letter,
			final java.math.BigDecimal count_0,
			final java.math.BigDecimal count_1,
			final java.math.BigDecimal count_2,
			final java.math.BigDecimal count_3
	) {
		this.initial = initial;
		this.letter = letter;
		this.count_0 = count_0;
		this.count_1 = count_1;
		this.count_2 = count_2;
		this.count_3 = count_3;
	}

	public Probability(
			final Character initial, final Character letter,
			final int count_0,
			final int count_1,
			final int count_2,
			final int count_3
	) {
		this.initial = initial.toString();
		this.letter = letter.toString();
		this.count_0 = new java.math.BigDecimal(count_0);
		this.count_1 = new java.math.BigDecimal(count_1);
		this.count_2 = new java.math.BigDecimal(count_2);
		this.count_3 = new java.math.BigDecimal(count_3);
	}
}
